aten/src/ATen/native/ReduceOps.cpp

```diff
@@ -1179,7 +1179,6 @@ Tensor& sum_out(const Tensor& self, DimnameList dim,
 
 Tensor& nansum_out(const Tensor& self, at::OptionalIntArrayRef dim,
                        bool keepdim, optional<ScalarType> opt_dtype, Tensor& result) {
-  TORCH_CHECK(!c10::isComplexType(self.scalar_type()), "nansum does not support complex inputs");
```

Comment:
Would it make sense to change it to the following so that CPU will still throw a nice error message.
```c++
if (self.device == 'cpu') {
   TORCH_CHECK(!c10::isComplexType(self.scalar_type()), "nansum does not support complex inputs");
}

Reply:
Sure!